bundles/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java

```diff
@@ -273,13 +293,17 @@ public void call(Object... args) {
                 .on(Socket.EVENT_RECONNECT_FAILED,
                         args -> logger.debug("Socket.IO re-connect attempts failed. Stopping reconnection."))//
                 .on(Socket.EVENT_DISCONNECT, args -> {
-                    if (args.length > 0) {
-                        logger.warn("Socket.IO disconnected: {}", args[0]);
-                    } else {
-                        logger.warn("Socket.IO disconnected");
-                    }
+                    String message = args.length > 0 ? args[0].toString() : "";
+                    logger.warn("Socket.IO disconnected: {}", message);
                     isConnected = false;
                     onDisconnect();
+                    // https://github.com/socketio/socket.io-client/commit/afb952d854e1d8728ce07b7c3a9f0dee2a61ef4e
+                    if ("io server disconnect".equals(message)) {
```

Comment:
@ssalonen last change, after more testing i can confirm when the server calls disconnect, our client will not reconnect.  This is actually a design flaw (in my opinion) of the socket.io client library.  You can see in the code comment that this is actually what the official docs suggest doing, checking for this string.  Hopefully this solves yet another reconnect issue !   I have done quite a bit of testing on this and feel good about this change (along with the thread safe scheduling of course)

Reply:
What a find! Really good